kimera_vio_ros/src/interfaces/base_interface.cpp

```diff
@@ -0,0 +1,74 @@
+#include <chrono>
+
+#include "kimera_vio_ros/interfaces/base_interface.hpp"
+
+using namespace std::chrono_literals;
+
+namespace kimera_vio_ros
+{
+namespace interfaces
+{
+
+BaseInterface::BaseInterface(
+  rclcpp::Node::SharedPtr & node)
+: VIO::DataProviderInterface(),
+  node_(node),
+  vio_params_(nullptr),
+  vio_pipeline_(nullptr),
+  tf_buffer_(nullptr),
+  tf_broadcaster_(nullptr),
+  tf_listener_(nullptr)
+{
+  tf_buffer_ = std::make_shared<tf2_ros::Buffer>(node_->get_clock());
+  tf_broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(node_);
+  tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_, node_, false);
+
+  callback_group_pipeline_ = node_->create_callback_group(
+    rclcpp::callback_group::CallbackGroupType::MutuallyExclusive);
+
+  base_link_frame_id_ = node_->declare_parameter(
+    "frame_id.base_link", "base_link");
+  map_frame_id_ = node_->declare_parameter(
+    "frame_id.map", "map");
+  world_frame_id_ = node_->declare_parameter(
+    "frame_id.world", "world");
+
+  std::string params_folder_;
+  params_folder_ = node_->declare_parameter(
+    "params_folder", "");
+  CHECK(!params_folder_.empty());
+  vio_params_ = std::make_shared<VIO::VioParams>(
```

Comment:
@ToniRV After the recent changes to the front end, it looks like vio_params_ has been made separate from VIO::DataProviderInterface class. This might make it easier to work with when using stuff like camera_info topics to init the extrinsic and intrinsic camera parameters, so thanks. 

However, it looks like it seeks a number of yaml files with hardcoded file names, like the left and right camera yamls, but also some new files that where never added to the realsense data set params folder that would include a reference to the default hyperparameters for that device:

* https://github.com/MIT-SPARK/Kimera-VIO/tree/83e12698a5803dcb6e6111dbed25b7ef59c878bb/params/RealSense

Could the reference params folder for the RealSense be updated to re-sync with the recent refactoring updtream?